core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1002,15 +977,19 @@ public void parseXContent(QueryParseContext context, AggregatorParsers aggParser
                         scriptFields.add(new ScriptField(context));
                     }
                 } else if (context.getParseFieldMatcher().match(currentFieldName, INDICES_BOOST_FIELD)) {
-                    indexBoost = new ObjectFloatHashMap<>();
+                    if (!indexBoosts.isEmpty()) {
```

Comment:
no there was no misunderstanding, I just didn't realize the implications of this check